ReactWindows/ReactNative.Shared/UIManager/DependencyObjectExtensions.cs

```diff
@@ -14,6 +14,7 @@ namespace ReactNative.UIManager
     /// </summary>
     public static class DependencyObjectExtensions
     {
+        // ConditionalWeakTable is thread safe
         private static readonly ConditionalWeakTable<DependencyObject, DependencyObjectData> s_properties =
```

Comment:
This brings up a good point - since we have reliable hooks to clean up after the DependencyObject is removed from the React hierarchy, should we just switch to a concurrent dictionary?

Reply:
If the magical self "cleaning" of ConditionalWeakTable was the only reason to use it, I guess yes, I could use a concurrent dictionary. It's true, we do seem to cover all cleanup cases through that DropView hook.

---
In reply to: [165942172](https://github.com/Microsoft/react-native-windows/pull/1615#discussion_r165942172) [](ancestors = 165942172)